ramiro_serra/color_sensor_notes.md

```diff
@@ -0,0 +1,169 @@
+
+Now, let's check if it's working.
+
+    roslaunch ca_gazebo create_autorace.launch
+
+We can see the updated model in `gazebo`:
+
+![gazeboupdated](media/gazebo_model.png)
+
+But in `rviz`, the model is a little buggy:
+
+![rvizbug](media/rviz_bug.png)
+
+That's because the `fixed frame` is set to `/map`, a frame that is not defined in our case.
+So set the `fixed frame` to `base_link` for example, and now it shows correctly:
+
+![rviznice](media/rviz_nice.png)
+
+## Manually converting from xacro --> URDF --> SDF
+
+We can check this info in the `README.md` file inside the `ca_description` package.
+
+Let's give it a try:
+
+    roscd ca_description/urdf/
+    roscore &
+    rosrun xacro xacro --inorder create_2.xacro visualize:=false > test.urdf
+
+We can check it created a `test.urdf` file.
+To check whether the sintax is fine or has errors:
+
+    check_urdf test.urdf
+
+
+    robot name is: create_2
+    ---------- Successfully Parsed XML ---------------
+    root Link: base_footprint has 1 child(ren)
+        child(1):  base_link
+            child(1):  left_color_sensor
+            child(2):  right_color_sensor
+            child(3):  wall_sensor_link
+            child(4):  block_link
+                child(1):  laser_link
+                child(2):  raspicam_link
+                    child(1):  raspicam_optical_frame
+            child(5):  caster_wheel_link
+            child(6):  front_left_cliff_sensor_link
+            child(7):  front_right_cliff_sensor_link
+            child(8):  imu_link
+            child(9):  left_wheel_link
+            child(10):  right_wheel_link
+            child(11):  side_left_cliff_sensor_link
+            child(12):  side_right_cliff_sensor_link
+
+And now we can manually convert to SDF.
+To see how Gazebo will use the robot description:
+
+    gz sdf -p test.urdf > test.sdf
+
+You can open the `test.sdf` file with whatever editor you feel like, and check what's going to be used by gazebo.
+
+## Adding optical frame
+
+We need to create a link where our camera/sensor will be attached to.
+In order to keep by the `rosREP` standard, we use the axis they suggest:
+
+    Suffix Frames
+    In the case of cameras, there is often a second frame defined with a "_optical" suffix. 
+    This uses a slightly different convention:
+
+    z forward
+    x right
+    y down
+
+So, in our `ramiro_color_sensor.xacro`, we add:
+
+    <joint name="${prefix}_color_optical_joint" type="fixed">
+      <origin xyz="0 0 0" rpy="${-pi/2} 0 ${-pi/2}" />
+      <parent link="${prefix}_color_sensor" />
+      <child link="${prefix}_color_optical_frame"/>
+    </joint>
+
+    <link name="${prefix}_color_optical_frame"/>
+
+Now, `roslaunch ca_gazebo create_autorace.launch`:
+
+And we can check in `rviz`, selecting the `tf's` we want to check, that the axis are correctly placed:
+
+![optical_frame](media/optical_frame_axis.png).
```

Comment:
That axis is giant compared to the model. I don't know if it can be done but, could you decrease the size of the axis?

Reply:
I read that the axis should be normalized, the thing is that, being the measuring unit `meters`, the axis will be 1m long, and it's a lot larger than the model's size. I could change it, but i don't think it's a good practice.

Reply:
I meant the [visualization axis in RViz](http://wiki.ros.org/rviz/DisplayTypes/Axes). You're talking about [this](http://wiki.ros.org/urdf/XML/joint#Elements).